pkg/webhooks/machine_webhook.go

```diff
@@ -487,6 +488,12 @@ func MachineSetMutatingWebhook() admissionregistrationv1.MutatingWebhook {
 }
 
 func (h *machineValidatorHandler) validateMachine(m, oldM *machinev1.Machine) (bool, []string, utilerrors.Aggregate) {
+	// Skip validation if we just remove the finalizer.
```

Comment:
You need to check that everything else is the same no? This change allows any change when removing the finalizer right?